Trouble while compiling

Hi !

I'm trying to build your project to help a friend of mine.

Unfortunately, I couldn't manage to compile it, despite the readme indications.

I'm trying to build it with :
- debian bookworm
- cmake 3.25.1
- clang 14.0.6 (debian version)
- g++ 12.2.0
- libcgal-dev  5.5.1-2

Can you give me details on the configuration you are using to compile your project ?
Do you have any non default configuration for the tools you are using ?

Best regards.

Sure !

Here is the log for the cmake command :

```
root@6d9a5409476e:/workspaces/YaoGraph/build# cmake ../-DWITH_CMAKE=ON ../
CMake Warning:
  Ignoring extra path from command line:

   "/workspaces/YaoGraph/-DWITH_CMAKE=ON"


-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Defaulting CMAKE_BUILD_TYPE to Release
-- Configuring done
-- Generating done
-- Build files have been written to: /workspaces/YaoGraph/build
```

And for the make command :

```
root@6d9a5409476e:/workspaces/YaoGraph/build# make 
[ 25%] Building CXX object CMakeFiles/YaoGraph.dir/YaoGraph.cpp.o
/workspaces/YaoGraph/YaoGraph.cpp: In function 'auto runAlg(const tDim&, const tPoints&, const tBox&, Args ...)':
/workspaces/YaoGraph/YaoGraph.cpp:100:67: error: 'KernelName' was not declared in this scope; did you mean 'Kernel'?
  100 |                       << Algorithm<Kernel>::name() << " with " << KernelName<Kernel>::name() << " kernel" << std::endl;
      |                                                                   ^~~~~~~~~~
      |                                                                   Kernel
/workspaces/YaoGraph/YaoGraph.cpp:100:84: error: expected primary-expression before '>' token
  100 |                       << Algorithm<Kernel>::name() << " with " << KernelName<Kernel>::name() << " kernel" << std::endl;
      |                                                                                    ^
/workspaces/YaoGraph/YaoGraph.cpp:100:87: error: '::name' has not been declared; did you mean 'tzname'?
  100 |                       << Algorithm<Kernel>::name() << " with " << KernelName<Kernel>::name() << " kernel" << std::endl;
      |                                                                                       ^~~~
      |                                                                                       tzname
/workspaces/YaoGraph/YaoGraph.cpp:108:40: error: 'KernelName' was not declared in this scope; did you mean 'Kernel'?
  108 |                       << " kernel=" << KernelName<Kernel>::name()
      |                                        ^~~~~~~~~~
      |                                        Kernel
/workspaces/YaoGraph/YaoGraph.cpp:108:57: error: expected primary-expression before '>' token
  108 |                       << " kernel=" << KernelName<Kernel>::name()
      |                                                         ^
/workspaces/YaoGraph/YaoGraph.cpp:108:60: error: '::name' has not been declared; did you mean 'tzname'?
  108 |                       << " kernel=" << KernelName<Kernel>::name()
      |                                                            ^~~~
      |                                                            tzname
make[2]: *** [CMakeFiles/YaoGraph.dir/build.make:76: CMakeFiles/YaoGraph.dir/YaoGraph.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/YaoGraph.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

And juste to be sure, Here is the log without the cmake argument :

```
[root@6d9a5409476e:/workspaces/YaoGraph/build# cmake ../
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Defaulting CMAKE_BUILD_TYPE to Release
-- Configuring done
-- Generating done
-- Build files have been written to: /workspaces/YaoGraph/build]([url](url))
```

With the same error :

```
root@6d9a5409476e:/workspaces/YaoGraph/build# make
[ 25%] Building CXX object CMakeFiles/YaoGraph.dir/YaoGraph.cpp.o
/workspaces/YaoGraph/YaoGraph.cpp: In function 'auto runAlg(const tDim&, const tPoints&, const tBox&, Args ...)':
/workspaces/YaoGraph/YaoGraph.cpp:100:67: error: 'KernelName' was not declared in this scope; did you mean 'Kernel'?
  100 |                       << Algorithm<Kernel>::name() << " with " << KernelName<Kernel>::name() << " kernel" << std::endl;
      |                                                                   ^~~~~~~~~~
      |                                                                   Kernel
/workspaces/YaoGraph/YaoGraph.cpp:100:84: error: expected primary-expression before '>' token
  100 |                       << Algorithm<Kernel>::name() << " with " << KernelName<Kernel>::name() << " kernel" << std::endl;
      |                                                                                    ^
/workspaces/YaoGraph/YaoGraph.cpp:100:87: error: '::name' has not been declared; did you mean 'tzname'?
  100 |                       << Algorithm<Kernel>::name() << " with " << KernelName<Kernel>::name() << " kernel" << std::endl;
      |                                                                                       ^~~~
      |                                                                                       tzname
/workspaces/YaoGraph/YaoGraph.cpp:108:40: error: 'KernelName' was not declared in this scope; did you mean 'Kernel'?
  108 |                       << " kernel=" << KernelName<Kernel>::name()
      |                                        ^~~~~~~~~~
      |                                        Kernel
/workspaces/YaoGraph/YaoGraph.cpp:108:57: error: expected primary-expression before '>' token
  108 |                       << " kernel=" << KernelName<Kernel>::name()
      |                                                         ^
/workspaces/YaoGraph/YaoGraph.cpp:108:60: error: '::name' has not been declared; did you mean 'tzname'?
  108 |                       << " kernel=" << KernelName<Kernel>::name()
      |                                                            ^~~~
      |                                                            tzname
make[2]: *** [CMakeFiles/YaoGraph.dir/build.make:76: CMakeFiles/YaoGraph.dir/YaoGraph.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/YaoGraph.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

And if this is of any help, I'm running this from a docker container, from the following Dockerfile :

```
FROM debian:bookworm

RUN apt update

RUN apt install -y make cmake g++ gcc clang libcgal-dev

RUN ["/bin/bash"]
```

Best regards

--- a/include/Kernels/CGALKernel.hpp
+++ b/include/Kernels/CGALKernel.hpp
@@ -19,13 +19,6 @@
 using ExactPredicatesInexactConstructions = CGAL::Exact_predicates_inexact_constructions_kernel;
 using ExactPredicatesExactConstructions = CGAL::Exact_predicates_exact_constructions_kernel;
 
-template<typename K>
-struct KernelName {
-    static std::string name() {
-        return K::name();
-    }
-};
-
 template<>
 struct KernelName<ExactPredicatesExactConstructions> {
     static std::string name() {

--- a/include/Types.hpp
+++ b/include/Types.hpp
@@ -106,3 +106,10 @@ struct tYaoVertex {
 struct tYaoGraph : public std::vector<tYaoVertex> {
     tYaoGraph(const tIndex &n, const tDim &k) : std::vector<tYaoVertex>(n, tYaoVertex(k)) {}
 };
+
+template<typename K>
+struct KernelName {
+    static std::string name() {
+        return K::name();
+    }
+};

